### PR TITLE
Change v1beta1 apis

### PR DESCRIPTION
**Pull request by zmberg**

### CODE REVIEW

File: api/v1beta1/rollout_types.go [hunk 1]
```diff
@@ -122,6 +102,30 @@ type CanaryStrategy struct {
...
 	// +optional
 	PatchPodTemplateMetadata *PatchPodTemplateMetadata `json:"patchPodTemplateMetadata,omitempty"`
+	// End-to-end progressive delivery capacity for microservice application,
+	// Can only be used in conjunction with the batches field, such as:
+	// trafficRouting: spring-cloud
+	// batches:
+	// - replicas: 2
+	// This means constructing end-to-end grayscale environment,
+	// grayscale instance 2, where the traffic policy is trafficRouting
+	TrafficRouting string `json:"trafficRouting,omitempty"`
```
Comment (furykerry) [inline api/v1beta1/rollout_types.go:112]: TrafficRouting string should be a field in CanaryStrategy

File: api/v1beta1/rollout_types.go [hunk 2]
```diff
@@ -101,17 +81,17 @@ type RolloutStrategy struct {
...
+	// - replicas: 1  # batches 0
+	// - replicas: 2  # batches 1
+	// - replicas: 5  # batches 2
+	// Not that these canaryReplicas should be a non-decreasing sequence.
 	// +optional
-	Steps []CanaryStep `json:"steps,omitempty"`
-	// TrafficRoutings hosts all the supported service meshes supported to enable more fine-grained traffic routing
-	// and current only support one TrafficRouting
-	TrafficRoutings []v1alpha1.TrafficRoutingRef `json:"trafficRoutings,omitempty"`
+	Batches []BatchStrategy `json:"batches,omitempty"`
```
Comment (furykerry) [inline api/v1beta1/rollout_types.go:94]: Batches should be pointer to BatchStrategy, and in BatchStrategy , one can define replicas for each batches (maybe a field called BatchStep), and other strategy configuration

File: api/v1beta1/rollout_types.go [hunk 3]
```diff
@@ -124,6 +105,29 @@ type CanaryStrategy struct {
...
+}
+
+// CanaryStrategy defines parameters for a Replica Based Canary
+type CanaryStrategy struct {
+	// Steps define the order of phases to execute release in batches(20%, 40%, 60%, 80%, 100%)
+	// +optional
+	Steps []CanaryStep `json:"steps"`
+	// TrafficRoutings hosts all the supported service meshes supported to enable more fine-grained traffic routing
+	// and current only support one TrafficRouting
+	TrafficRoutings []v1alpha1.TrafficRoutingRef `json:"trafficRoutings"`
```
Comment (veophi) [inline api/v1beta1/rollout_types.go:125]: what if users use `batch release` + `trafficRouting` ？

File: api/v1beta1/rollout_types.go [hunk 4]
```diff
@@ -124,6 +105,29 @@ type CanaryStrategy struct {
 	PatchPodTemplateMetadata *PatchPodTemplateMetadata `json:"patchPodTemplateMetadata,omitempty"`
 }
 
+type BatchStrategy struct {
+	BatchStep []BatchStep `json:"batchStep"`
```
Comment (furykerry) [inline api/v1beta1/rollout_types.go:109]: just name it steps in the json?

File: api/v1beta1/rollout_types.go [hunk 5]
```diff
@@ -101,17 +81,18 @@ type RolloutStrategy struct {
...
+	//   - replicas: 1  # batches 0
+	//   - replicas: 2  # batches 1
+	//   - replicas: 5  # batches 2
+	// Not that these canaryReplicas should be a non-decreasing sequence.
 	// +optional
-	Steps []CanaryStep `json:"steps,omitempty"`
-	// TrafficRoutings hosts all the supported service meshes supported to enable more fine-grained traffic routing
-	// and current only support one TrafficRouting
-	TrafficRoutings []v1alpha1.TrafficRoutingRef `json:"trafficRoutings,omitempty"`
+	Batches *BatchStrategy `json:"batches,omitempty"`
```
Comment (furykerry) [inline api/v1beta1/rollout_types.go:95]: batches -> Batch ?

**Pull request closed without merging**